reco_utils/recommender/ripplenet/model.py

```diff
@@ -0,0 +1,241 @@
+                     for gradient in gradients]
+        self.optimizer = optimizer.apply_gradients(zip(gradients, variables))
+        '''
+
+    def train(self, sess, feed_dict):
+        return sess.run([self.optimizer, self.loss], feed_dict)
+    
+    def return_scores(self, sess, feed_dict):
+        labels, scores = sess.run([self.labels, self.scores_normalized], feed_dict)
+        return labels, scores
+
+    def eval(self, sess, feed_dict):
+        labels, scores = sess.run([self.labels, self.scores_normalized], feed_dict)
+        auc = roc_auc_score(y_true=labels, y_score=scores)
+        predictions = [1 if i >= 0.5 else 0 for i in scores]
+        acc = np.mean(np.equal(predictions, labels))
+        return auc, acc
+    
+    def get_feed_dict(self, n_hop, data, ripple_set, start, end):
+        feed_dict = dict()
+        feed_dict[self.items] = data[start:end, 1]
+        feed_dict[self.labels] = data[start:end, 2]
+        for i in range(n_hop):
+            feed_dict[self.memories_h[i]] = [ripple_set[user][i][0] for user in data[start:end, 0]]
+            feed_dict[self.memories_r[i]] = [ripple_set[user][i][1] for user in data[start:end, 0]]
+            feed_dict[self.memories_t[i]] = [ripple_set[user][i][2] for user in data[start:end, 0]]
+        return feed_dict
+
+    def print_metrics_evaluation(self, sess, n_hop, data, ripple_set, batch_size):
+        start = 0
+        auc_list = []
+        acc_list = []
+        while start < data.shape[0]:
+            auc, acc = self.eval(sess, self.get_feed_dict(n_hop, 
+                                                         data, ripple_set,
+                                                          start, start + batch_size))
+            auc_list.append(auc)
+            acc_list.append(acc)
+            start += batch_size
+        return float(np.mean(auc_list)), float(np.mean(acc_list))
+
+    def fit(self,
+            sess, 
+            n_epoch, batch_size, n_hop,
+            train_data, ripple_set, show_loss):
+        sess.run(tf.global_variables_initializer())
+        for step in range(n_epoch):
+            # training
+            np.random.shuffle(train_data)
+            start = 0
+            while start < train_data.shape[0]:
+                _, loss = self.train(sess, 
+                                     self.get_feed_dict(n_hop, 
+                                     train_data, ripple_set, 
+                                     start, start + batch_size))
+                start += batch_size
+                if show_loss:
+                    print('%.1f%% %.4f' % (start / train_data.shape[0] * 100, loss))
+
+            train_auc, train_acc = self.print_metrics_evaluation(sess, n_hop,
+                                                                 train_data, ripple_set,
+                                                                 batch_size)
+
+            print('epoch %d  train auc: %.4f  acc: %.4f'
```

Comment:
in other parts of the code we are using loggers: https://github.com/microsoft/recommenders/blob/master/reco_utils/dataset/download_utils.py#L12